pennylane_lightning/core/src/lightning.kokkos.toml

```
@@ -0,0 +1,79 @@
+schema = 1
+
+[device]
+name = "lightning.kokkos"
+version = "v0.31.0"
+precision = ['float32', 'float64']
+
+[operations]
+observables = [
+	"NamedObs",
+	"HermitianObs",
+	"TensorObs",
+	"HamiltonianObs",
+]
+
+[[operations.gates]]
+full = [
+	"Identity",
+	"PauliX",
+	"PauliY",
+	"PauliZ",
+	"Hadamard",
+	"S",
+	"T",
+	"PhaseShift",
+	"RX",
+	"RY",
+	"RZ",
+	"Rot",
+	"CNOT",
+	"CY",
+	"CZ",
+	"SWAP",
+	"IsingXX",
+	"IsingXY",
+	"IsingYY",
+	"IsingZZ",
+	"ControlledPhaseShift",
+	"CRX",
+	"CRY",
+	"CRZ",
+	"CRot",
+	"Toffoli",
+	"CSWAP",
+	"MultiRZ",
```

Excitation gates have are `NamedOperation`s, but `MultiControlledX` does not have a specialized kernel and must be applied as a `QubitUnitary`.
```suggestion
	"MultiRZ",
        "SingleExcitation",
        "SingleExcitationMinus",
        "SingleExcitationPlus",
        "DoubleExcitation",
        "DoubleExcitationMinus",
        "DoubleExcitationPlus",
```

---

I see thanks!

This actually illustrates a misconception about what this file actually defines. This file is actually more related to catalyst than pennylane-lightning. In catalyst we have functions that map to `NamedOperation`. For example:

```cpp
void __quantum__qis__RX(double theta, QUBIT *qubit, bool adjoint)
{
    Catalyst::Runtime::CTX->getDevice()->NamedOperation("RX", {theta},
                                                        {reinterpret_cast<QubitIdType>(qubit)},
                                                        /* inverse = */ adjoint);
}
```

One interpretation (that I think is the correct interpretation) of this file is that the `qml.Operations` in the `full` field in the toml file are `qml.Operations` that are lowered to `__quantum__qis__${INSTRUCTION}` calls. And those gates under the `matrix` field in the toml file are to be lowered to `__quantum__qis__QubitUnitary` calls.

I think since `__quantum__qis__${INSTRUCTION}` function calls model the Quantum Instruction Set (hence QIS) it makes sense to add `__quantum__qis__SingleExcitation` and the like.